clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/DefaultClusteringService.java

```diff
@@ -272,35 +267,23 @@ public boolean handlesResourceType(ResourceType<?> resourceType) {
   @Override
   public PersistenceSpaceIdentifier getPersistenceSpaceIdentifier(String name, CacheConfiguration<?, ?> config) throws CachePersistenceException {
     DefaultClusterCacheIdentifier identifier = new DefaultClusterCacheIdentifier(name);
-    Tuple<DefaultClusterCacheIdentifier, ClusteredMapRepository> existing = knownPersistenceSpaces
-        .putIfAbsent(name, new Tuple<DefaultClusterCacheIdentifier, ClusteredMapRepository>(identifier, new ClusteredMapRepository()));
-    if (existing != null) {
-      identifier = existing.first;
-    }
     return identifier;
   }
 
-  private ConcurrentHashMap<EntityRef<ConcurrentClusteredMap, Object>, ConcurrentClusteredMap<?, ?>> getNewMapEntityMap() {
-    return new ConcurrentHashMap<EntityRef<ConcurrentClusteredMap, Object>, ConcurrentClusteredMap<?, ?>>();
-  }
-
   @Override
   public void releasePersistenceSpaceIdentifier(PersistenceSpaceIdentifier<?> identifier) throws CachePersistenceException {
-    if (!isKnownIdentifier(identifier)) {
+    ClusteredCacheIdentifier clusterCacheIdentifier = (ClusteredCacheIdentifier) identifier;
+    if (!knownPersistenceSpaces.containsKey(clusterCacheIdentifier)) {
       throw new CachePersistenceException("Unknown identifier: " + identifier);
     }
-    DefaultClusterCacheIdentifier clusterCacheIdentifier = (DefaultClusterCacheIdentifier) identifier;
-    Tuple<DefaultClusterCacheIdentifier, ClusteredMapRepository> tuple = knownPersistenceSpaces.remove(clusterCacheIdentifier.getId());
-    tuple.second.clear();
+    knownPersistenceSpaces.remove(clusterCacheIdentifier);
   }
 
   @Override
   public StateRepository getStateRepositoryWithin(PersistenceSpaceIdentifier<?> identifier, String name) throws CachePersistenceException {
-    if (!isKnownIdentifier(identifier)) {
-      throw new CachePersistenceException("Unknown identifier: " + identifier);
-    }
-    DefaultClusterCacheIdentifier clusterCacheIdentifier = (DefaultClusterCacheIdentifier) identifier;
-    return new ClusteredStateRepository(clusterCacheIdentifier, name, this);
+    ClusteredCacheIdentifier clusterCacheIdentifier = (ClusteredCacheIdentifier) identifier;
+    knownPersistenceSpaces.putIfAbsent(clusterCacheIdentifier, new ClusteredStateRepository(clusterCacheIdentifier, name, this.entity));
```

Comment:
This use of `putIfAbsent` is wrong - see [there](https://github.com/ehcache/ehcache3/blob/080be1433862d32fcaf485436a04f61f17417546/impl/src/main/java/org/ehcache/impl/serialization/TransientStateRepository.java#L38) for the proper pattern